proposals/20191030-api.md

```diff
@@ -0,0 +1,118 @@
+# Falco APIs
+
+## Summary
+
+This is a proposal to better structure the Falco API.
+
+It will contain a list of gRPC services that can be added to have a first stable version of the API for v1.0.0.
+
+## Motivation
+
+We want to enable users to use thirdy-party clients to interface with Falco outputs, inputs, rules, and configurations.
+
+Such ability would enable the community to create a whole set of OSS tools, built on top of Falco.
+
+Some examples, already in place, are:
+
+- [falcoctl](https://github.com/falcosecurity/falcoctl)
+- [falco-exporter](https://github.com/falcosecurity/falco-exporter)
+
+## Goals
+
+- To design and implement an API to make Falco more configurable
+- To design and implement an API to make Falco more consumable
+- To design a set of simple contract interfaces
+- To have a secure by default Falco API
+
+## Non-Goals
+
+- Replace the existing inputs: the **inputs** will be only for new source events users may want to add, initially
+
+## Proposal
+
+### Use cases
+
+- Receive Falco events with a well-defined contract over wire --> **outputs**
```

Comment:
remove wire? again this goes back to my previous comment about the API being a shape and agreement and not an implementation.

For instance if a user wanted to vendor falco in C++ they would still potentially be using "an API"